OpenRA.Mods.Common/Traits/Render/WithBuildingExplosion.cs

```diff
@@ -27,22 +27,21 @@ class WithBuildingExplosionInfo : ITraitInfo, Requires<BuildingInfo>
 		[Desc("Custom palette name")]
 		public readonly string Palette = "effect";
 
-		public object Create(ActorInitializer init) { return new WithBuildingExplosion(init.Self, this); }
+		public object Create(ActorInitializer init) { return new WithBuildingExplosion(this); }
 	}
 
 	class WithBuildingExplosion : INotifyKilled
 	{
 		WithBuildingExplosionInfo info;
-		BuildingInfo buildingInfo;
 
-		public WithBuildingExplosion(Actor self, WithBuildingExplosionInfo info)
+		public WithBuildingExplosion(WithBuildingExplosionInfo info)
 		{
 			this.info = info;
-			buildingInfo = self.Info.Traits.Get<BuildingInfo>();
```

Comment:
This one is safe, because it defines `Requires<BuildingInfo>` on WithBuildingExplosionInfo.